IA-1891: Planning view list tab sort by assignments

beygorghor opened the pull request:
Be able to sort org units by assignment column.

Related JIRA tickets : IA-1891

## Self proof reading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [ ] Did I add translations
- [ ] My migrations file are included
- [ ] Are there enough tests

## Changes

- adding filters in microplanning.py
- remove sortable:false from assignment column
- dynamic sort/order depending on current team type


## How to test

View a planning by clicking on a planning in list view
Select tab List
Make sure you have some assignents
Try to sort on the last column (with a team of users and a team of team)

## Print screen / video


https://user-images.githubusercontent.com/12494624/217469677-0716090c-bfa3-460d-9e2f-fb27867b98b7.mov

quang-le commented:
Enabled `OrgUnitTreeviewModal`'s `title` prop  to receive a string as well as an message object

mathvdh committed:
Remove sortable and fix api microplanning assignments

beygorghor committed:
Merge branch 'IA-1893-planning-fit-to-bounds' into IA-1891-planning-view-list-tab-be-able-to-sort-org-units-by-assignment-column

beygorghor committed:
enable sort on assignment

- change column id with currentTeam type
- useEffect to change order if currentTeam type is different
- invalidate org units queries

quang-le commented on hat/assets/js/apps/Iaso/domains/assignments/hooks/requests/useSaveAssignment.ts:
```diff
@@ -16,13 +16,14 @@ export const saveAssignment = async (
+export const useSaveAssignment = (): UseMutationResult => {
+    const queryClient = useQueryClient();
+    const onSuccess = () => {
+        queryClient.invalidateQueries('orgUnits');
+        queryClient.invalidateQueries('assignmentsList');
```
workaround to invalidate multiple query keys?

beygorghor replied:
yes, could not find another way

quang-le reviewed (approved):
LGTM
